### ISSUE TITLE
Vulnerability Not Detected

### ISSUE THREAD

**Issue by ayedo** [DependencyTrack/dependency-track#4267]
### Current Behavior

The vulnerability [GHSA-qj66-m88j-hmgj](https://github.com/advisories/GHSA-qj66-m88j-hmgj):

![Image](https://github.com/user-attachments/assets/0397a333-a7f4-4efe-a8e5-0af349b56b4b)

Is not detected for the following component:

![Image](https://github.com/user-attachments/assets/e47af3bb-4562-434b-8325-65d4de2029d7)

### Steps to Reproduce

1. Add the component shown in the image to a project
2. Enable GitHub advisories
3. Run analyser on the project with that component

### Expected Behavior

I expect the vulnerability to be detected, since the component version lies within the version range described by the GitHub advisory.

I saw that the versions are a bit special, thez contain "preview" and other punctuation - could it be related to that?

### Dependency-Track Version

4.12.0

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Database Server Version

16.4

### Browser

Mozilla Firefox

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [x] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by nscuro** [comment]
> I saw that the versions are a bit special, thez contain "preview" and other punctuation - could it be related to that?

Yeah it seems DT can't work out that `8.0.0-preview.1.23110.8` is lower than `8.0.0`. #2826 could help to deal with NuGet-specific versioning like this.

**Comment by ayedo** [comment]
@nscuro  thanks for your quick reply. I'm trying to figure out a work around to add the vulnerability manually. Do you have an idea how this could be achieved with reasonable work?

**Comment by antoinbo** [comment]
Hi @ayedo, have you found a workaround to detect those vulnerabilities?

**Comment by antoinbo** [comment]
I tried to reproduce the issue, but creating a component with the same PURL correctly found the vulnerability. I just found that I miss multiple GHSA for another component, and I wonder if this is due to GitHub API rate limitations?

**Comment by SA-KIRA** [comment]
I have the same issue
Missing some vulnerabilities
GHSA-jvhh-2m83-6w29 for example
